alpha/lib/model/entry.php

```diff
@@ -342,145 +346,90 @@ public function incViews ( $should_save = true )
-			switch ($field)
-			{
-				case self::NAME:
-					$originalValue = kString::alignUtf8String($defaultValue, self::MAX_NAME_LEN);
-					break;
-				case self::DESCRIPTION:
-					$originalValue = $defaultValue;
-					break;
-				case self::TAGS:
-					$originalValue =  ktagword::updateTags($this->tags, $defaultValue , $update_db );
-					break;
-			}
+			case $supportedFields['name']:
+				return kString::alignUtf8String($value, self::MAX_NAME_LEN);
+			case $supportedFields['description']:
+				return $value;
+			case $supportedFields['tags']:
+				return ktagword::updateTags($this->tags, $value , $update_db );
+			default:
+				return null;
 		}
-		
-		return array(self::IS_DEFAULT => !$isMapUpdated, self::DEFAULT_VALUE => $originalValue);
+	}
+	
+	public function setObjectDefaultLanguage($v)
+	{
+		$this->putInCustomData(self::ENTRY_DEFAULT_LANGUAGE, $v);
+	}
+	
+	public function getObjectDefaultLanguage()
+	{
+		return $this->getFromCustomData(self::ENTRY_DEFAULT_LANGUAGE, null, 'EN');
 	}
 	
 	public function setName($v)
 	{
-		PeerUtils::setExtension($this, $v, self::MAX_NAME_LEN, __FUNCTION__);
-		$multiLingualResult = $this->handleMultiLanguageInput(self::NAME, $v);
-		if ($multiLingualResult[self::IS_DEFAULT])
-		{
-			return parent::setName($multiLingualResult[self::DEFAULT_VALUE]);
-		}
-		return;
+		$multiLingualResult = multiLingualUtils::handleMultiLanguageInput($this, self::NAME, $v);
+		PeerUtils::setExtension($this, $multiLingualResult, self::MAX_NAME_LEN, __FUNCTION__);
+		return parent::setName($multiLingualResult);
 	}
 	
 	public function setDescription ($v)
 	{
-		$multiLingualResult = $this->handleMultiLanguageInput(self::DESCRIPTION, $v);
-		if ($multiLingualResult[self::IS_DEFAULT])
-		{
-			return parent::setDescription($multiLingualResult[self::DEFAULT_VALUE]);
-		}
-		return;
+		$multiLingualResult = multiLingualUtils::handleMultiLanguageInput($this, self::DESCRIPTION, $v);
+		
+		return parent::setDescription($multiLingualResult);
 	}
 	
-	protected function setValueByLanguageInMultiLangMapping(&$multiLangMapping, $field, $value, $language = null)
+	public function setTags($tags , $update_db = true )
 	{
-		if (!$language )
-		{
-			foreach ($value as $languageKey => $languageValue)
-			{
-				if (isset($multiLangMapping[$languageKey]))
-				{
-					unset($multiLangMapping[$languageKey]);
-				}
-			}
-			if(count($value) > 0)
-			{
-				$multiLangMapping[$field] = $value;
-			}
-		}
-		elseif ($multiLangMapping[$field][$language] != $value)
-		{
-			$multiLangMapping[$field][$language] = $value;
-			$this->setMultiLanguageMapping(json_encode($multiLangMapping, true));
-		}
+		$multiLingualResult = multiLingualUtils::handleMultiLanguageInput($this, self::TAGS, $tags);
+		return parent::setTags($multiLingualResult);
 	}
 
 	public function getName()
 	{
-		return parent::getName() . PeerUtils::getExtension($this, __FUNCTION__);
+		$defaultName = parent::getName() . PeerUtils::getExtension($this, __FUNCTION__);
```

Comment:
i would prefer us not to touch this function as nothing in the logic was changed (same for description and tags)